src/cpp/session/modules/SessionConnections.R

```diff
@@ -215,26 +215,69 @@ options(connectionObserver = list(
    NULL
 })
 
-.rs.addFunction("connectionReadSnippets", function() {
+.rs.addFunction("connectionFilesPath", function() {
    snippetsPath <- getOption("connections-path", "/etc/rstudio/connections/")
-   snippetsFiles <- list()
 
    if (!is.null(getOption("connections-path")) && !dir.exists(snippetsPath)) {
       warning(
          "Path '", snippetsPath, "' does not exist. ",
          "Configure the connections-path option appropriately.")
    }
+
+   snippetsPath
+})
+
+.rs.addFunction("connectionOdbcInstallerPath", function() {
+   normalizePath(
+      file.path(
+         .Call("rs_connectionOdbcInstallPath"),
+         "odbc",
+         "installers"),
+      mustWork = FALSE
+   )
+})
+
+.rs.addFunction("connectionFiles", function(include, defaultPath) {
+   connectionFiles <- list()
    
-   if (!is.null(snippetsPath)) {
-      snippetsFiles <- list.files(snippetsPath)
+   if (!is.null(defaultPath)) {
+      connectionFiles <- list.files(defaultPath)
    }
 
-   snippets <- lapply(snippetsFiles, function(file) {
-      fullPath <- file.path(snippetsPath, file)
-      paste(readLines(fullPath), collapse = "\n")
+   files <- lapply(connectionFiles, function(file) {
+      fullPath <- file.path(defaultPath, file)
    })
 
-   names(snippets) <- tools::file_path_sans_ext(snippetsFiles)
+   names(files) <- gsub(include, "", connectionFiles)
```

Comment:
Do we need to worry about case here?

Reply:
Not really, connection installers are only crafted by advanced users, so they can get it right rather than polluting with files named `.dCf`, `.dcF`, etc.